lock-api/src/test/java/com/palantir/lock/client/ConjureLockRequestTest.java

```diff
@@ -0,0 +1,131 @@
+    private static final String OLD_CONJURE_LOCK_REQUEST_FILE = "old";
+    private static final String WITH_METADATA_FILE = "with-metadata";
+    private static final boolean REWRITE_JSON_BLOBS = false;
+
+    private static final ConjureLockDescriptor LOCK_1 = ConjureLockDescriptor.of(Bytes.from(PtBytes.toBytes("abc")));
+    private static final ConjureLockDescriptor LOCK_2 = ConjureLockDescriptor.of(Bytes.from(PtBytes.toBytes("def")));
+    private static final ConjureLockDescriptor LOCK_3 = ConjureLockDescriptor.of(Bytes.from(PtBytes.toBytes("ghi")));
+    private static final ConjureLockDescriptor LOCK_4 = ConjureLockDescriptor.of(Bytes.from(PtBytes.toBytes("jkl")));
+    private static final byte[] BYTES_OLD = PtBytes.toBytes("old");
+    private static final byte[] BYTES_NEW = PtBytes.toBytes("new");
+    private static final byte[] BYTES_DELETED = PtBytes.toBytes("deleted");
+    private static final byte[] BYTES_CREATED = PtBytes.toBytes("created");
+    // The checksum on this does not have to be correct since the integrity check is performed by the server and not
+    // by Conjure itself
+    private static final ConjureLockRequestMetadata CONJURE_LOCK_REQUEST_METADATA = ConjureLockRequestMetadata.of(
+            ImmutableMap.of(
+                    0,
+                    ConjureChangeMetadata.unchanged(ConjureUnchangedChangeMetadata.of()),
+                    1,
+                    ConjureChangeMetadata.updated(
+                            ConjureUpdatedChangeMetadata.of(Bytes.from(BYTES_OLD), Bytes.from(BYTES_NEW))),
+                    2,
+                    ConjureChangeMetadata.deleted(ConjureDeletedChangeMetadata.of(Bytes.from(BYTES_DELETED))),
+                    3,
+                    ConjureChangeMetadata.created(ConjureCreatedChangeMetadata.of(Bytes.from(BYTES_CREATED)))),
+            ConjureLockDescriptorListChecksum.of(0, Bytes.from(PtBytes.toBytes("test-checksum-value"))));
+    private static final ConjureLockRequest CONJURE_LOCK_REQUEST = ConjureLockRequest.builder()
+            .requestId(new UUID(1337, 42))
+            .lockDescriptors(ImmutableList.of(LOCK_1, LOCK_2, LOCK_3, LOCK_4))
+            .acquireTimeoutMs(100)
+            .clientDescription("client: test, thread: test")
+            .build();
+
+    // AtlasDB (Client) serializes and TimeLock (Server) deserializes ConjureLockRequest objects.
+    // These are the respective mappers used internally by Conjure.
+    private static final ObjectMapper SERIALIZATION_MAPPER =
+            ObjectMappers.newClientObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    private static final ObjectMapper DESERIALIZATION_MAPPER =
+            ObjectMappers.newServerObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    // This mapper is used to ensure that two JSONs are equal excluding indentation
+    private static final ObjectMapper VERIFYING_MAPPER = new ObjectMapper();
+
```

Comment:
Yeah I thought about this and found it acceptable in the `LockEvent` case since copying the class was trivial, but copying the generated class is a different beast. Given that we will not have product dependencies, though, I think it is acceptable.

Reply:
I don't think it's the worst thing in the world for us to have a copy in the `test` folder here of the old-generated class. Granted, this is something we probably want anyways, as we have an invariant here which we must maintain, albeit super hard to break (our new format must be serializable by the old format). 

Even though this becomes out-dated after this PR, we still allow the upgrade path from a really old version of timelock to a really new version of timelock in the future, so it's useful we have this test to ensure we don't break anything until we `checkpoint` timelock.

Reply:
Sounds reasonable, I noticed that the test class is not really in the right module (should be co-located with `timelock-api.yml`), will move it